pyquil/operator_estimation.py

```diff
@@ -452,26 +596,24 @@ def _max_tpb_overlap(tomo_expt: TomographyExperiment):
         found_tpb = False
         # loop through dict items
         for es, es_list in diagonal_sets.items():
-            # update the dict value if es is diagonal in the same tpb as expt_setting
-            if _expt_settings_diagonal_in_tpb(es, expt_setting):
+            trial_es_list = es_list + [expt_setting]
+            diag_in_term = _max_weight_state(expst.in_state for expst in trial_es_list)
+            diag_out_term = _max_weight_operator(expst.out_operator for expst in trial_es_list)
+            if diag_in_term is not None and diag_out_term is not None:
```

Comment:
If `diag_in_term` and `diag_out_term` weren't recycled for good use a few lines later, I would've advocated for the use of an updated private function `_expt_settings_diagonal_in_tpb` (which seems to have been deprecated here). I would just add a comment here stating that this set of code is testing to see if both experiments can be carried out simultaneously/are diagonal in the same natural tpb (ntpb?)

Reply:
The comment below (`# shared tpb was found`) pretty much covers this, no?

Reply:
wrote a beefier comment

Reply:
looks good